examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/AuthenticationTest.kt

```diff
@@ -209,13 +212,51 @@ class AuthenticationTest: RealmTest() {
     }
 
     @Test
-    fun logoutTest() {
-        val app: App = App.create(YOUR_APP_ID) // Replace this with your App ID
-        runBlocking { // use runBlocking sparingly -- it can delay UI interactions
-            val user = app.login(Credentials.anonymous())
-            // :snippet-start: log-out
-            user.logOut()
+    fun customUserDataTest() {
+        val app: App = App.create(FLEXIBLE_APP_ID)
+        runBlocking {
+            app.login(Credentials.anonymous())
+            // :snippet-start: read-custom-user-data
+            val user = app.currentUser!!
+            val customUserData = user.customDataAsBsonDocument()
+            // :snippet-end:
+
+            // :snippet-start: refresh-custom-user-data
+            // Returns the updated custom data object
+            val updatedCustomData = user.refreshCustomData()
```

Comment:
`refreshCustomData` does not return any value, it just updates the custom data cached value that is stored internally by Realm.

The reason behind this decision is that `refreshCustomData` is a suspend function as it has to perform some network request to fetch the updated custom data and `customDataAsBsonDocument` is not, as it has direct access to the cached data.

```suggestion
            user.refreshCustomData()
            
            val userData = user.customDataAsBsonDocument()
            displayUserName(userData[USER_NAME]) 
```